### PR TITLE
Allow passing kwargs down to K.function

### PR DESCRIPTION
**Pull request by matt-peters**
This PR allows the user to pass `kwargs` down to `K.function` when calling `model.compile`.  For the Theano backend, there are useful compile time arguments that are not available via environmental variables or other means.  I don't know if there are equivalent flags for Tensorflow and have ignored them in this case.

Specifically, my use case needs to set `on_unused_input` to something other then `'raise'` (see http://deeplearning.net/software/theano/library/compile/function.html#function.function).  I'm using the `Graph` API and have loss functions that don't depend on `y_true`.  I'd also like to expose the results of intermediate layers in the final output without involving them in the final loss function.  In both cases, Theano throws a compile time error if `on_unused_input='raise'` (the default).

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -402,7 +402,7 @@ def __call__(self, inputs):
         return updated[:len(self.outputs)]
 
 
-def function(inputs, outputs, updates=[]):
+def function(inputs, outputs, updates=[], **kwargs):
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Since no arguments are expected, please add a check that none are passed (with helpful error message otherwise).

File: keras/backend/theano_backend.py [hunk 2]
```diff
@@ -449,8 +449,8 @@ def __call__(self, inputs):
         return self.function(*inputs)
 
 
-def function(inputs, outputs, updates=[]):
-    return Function(inputs, outputs, updates=updates)
+def function(inputs, outputs, updates=[], **kwargs):
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: Please add a check that all kwargs match a list of expected Theano arguments, with helpful error message otherwise. This is necessary to avoid confusion in case of typo.

**Comment by matt-peters** [comment]
Added error messages for both backends.  In Tensorflow case, I'm only warning that `kwargs` will be ignored instead of raising an exception (perhaps it's advantageous to allow the same code to be switched between backends without changing these?).  Can change to raise an exception if you prefer.

**Comment by fchollet** [comment]
LGTM

**Comment by matt-peters** [comment]
Thanks for the super fast review and merge :+1: This is an awesome library, it's nice to use, thank you!!
